app/modules/intelligence/agents_copy/chat_agents/crewai_rag_agent.py

```diff
@@ -0,0 +1,170 @@
+import json
+from typing import Any, List, Optional, AsyncGenerator
+from app.modules.intelligence.provider.provider_service import (
+    ProviderService,
+    AgentType,
+)
+from crewai import Agent, Crew, Process, Task
+from pydantic import BaseModel
+from app.modules.utils.logger import setup_logger
+from ..chat_agent import ChatAgent, ChatAgentResponse
+
+logger = setup_logger(__name__)
+
+
+class TaskConfig(BaseModel):
+    """Model for task configuration from agent_config.json"""
+
+    description: str
+    expected_output: str
+
+
+class AgentConfig(BaseModel):
+    """Model for agent configuration from agent_config.json"""
+
+    role: str
+    goal: str
+    backstory: str
+    tasks: List[TaskConfig]
+    max_iter: int = 5
+
+
+class CrewAIRagAgent(ChatAgent):
```

Comment:
I dont like the name CrewAIRagAgent, unfortunately I don't have suggestions either

Reply:
CrewAI and RAG both i dont want to include in the name